applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_static_shifted_boundary_solver.py

```diff
@@ -0,0 +1,67 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication import structural_mechanics_static_solver
+
+def CreateSolver(main_model_part, custom_settings):
+    return StaticMechanicalShiftedBoundarySolver(main_model_part, custom_settings)
+
+class StaticMechanicalShiftedBoundarySolver(structural_mechanics_static_solver.StaticMechanicalSolver):
+
+    def __init__(self, main_model_part, custom_settings):
+        # Construct the base solver and validate the remaining settings in the base class
+        super().__init__(main_model_part, custom_settings)
+
+        # Overwrite the base solver minimum buffer size
+        self.min_buffer_size = 1
+
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Construction finished")
+
+    @classmethod
+    def GetDefaultParameters(cls):
+        this_defaults = KratosMultiphysics.Parameters(r"""{
+            "conforming_basis" : true,
+            "extension_operator_type" : "MLS",
+            "mls_extension_operator_order" : 1
+        }""")
+        this_defaults.AddMissingParameters(super().GetDefaultParameters())
+        return this_defaults
+
+    def AddVariables(self):
+        # Add structural mechanics required variables
+        super().AddVariables()
+
+        # Add DISTANCE variable to represent the skin
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
+
+    def Initialize(self):
+        # Avoid zeros with positive epsilon
+        tol = 1.0e-12
+        for node in self.GetComputingModelPart().Nodes:
+            dist = node.GetSolutionStepValue(KratosMultiphysics.DISTANCE)
+            if abs(dist) < tol:
+                node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, 0, tol)
```

Comment:
variable utils

Reply:
The problem in here is the `if` statement (as far as I understand we cannot pass the function to be evaluated). Indeed, I'd do an SBM version of the distance modification process (I left this for a future PR).